File: src/mitris/sim/core/simulation/SimulationClock.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package mitris.sim.core.simulation;

/**
 *
 * @author jlrisco
 */
public class SimulationClock {

    protected double time;

    public SimulationClock(double time) {
        this.time = time;
    }

    public SimulationClock() {
        this(0);
    }

    public double getTime() {
        return time;
    }
    
    public void setTime(double time) {
        this.time = time;
    }
}

File: simulation/simulation_clock.go
package simulation

type SimulationClock interface {
	GetTime() float64
	SetTime(time float64)
}

func NewSimulationClock(t float64) SimulationClock {
	c := simulationClock{t}
	return &c
}

type simulationClock struct {
	time float64
}

func(c *simulationClock) GetTime() float64 {
	return c.time
}

func(c *simulationClock) SetTime(time float64) {
	c.time = time
}
